docs/tutorial/quickstart.md

```diff
@@ -0,0 +1,282 @@
+
+Before pushing a manifest, the layers it references must already exist in the repository. The following code snippet demonstrates how to push a config layer and a manifest layer using the [(*Repository) Push](https://pkg.go.dev/oras.land/oras-go/v2@v2.5.0/registry/remote#Repository.Push) method.
+
+```
+// push config layer
+configLayer := []byte("example config layer")
+configLayerDescriptor := content.NewDescriptorFromBytes(ocispec.MediaTypeImageLayer, configLayer)
+err = repo.Push(ctx, configLayerDescriptor, bytes.NewReader(configLayer))
+if err != nil {
+	panic(err)
+}
+fmt.Println("Pushed config layer")
+
+// push manifest layer
+manifestLayer := []byte("example manifest layer")
+manifestLayerDescriptor := content.NewDescriptorFromBytes(ocispec.MediaTypeImageLayer, manifestLayer)
+err = repo.Push(ctx, manifestLayerDescriptor, bytes.NewReader(manifestLayer))
+if err != nil {
+	panic(err)
+}
+fmt.Println("Pushed manifest layer")
+```
+
+## Push a manifest to the repository with the tag "quickstart"
+
+The following code snippet demonstrates how to assemble a manifest and push it to the repository with the tag "quickstart" using the [(*Repository) PushReference](https://pkg.go.dev/oras.land/oras-go/v2@v2.5.0/registry/remote#Repository.PushReference) method.
+
+```
+tag := "quickstart"
+manifest := ocispec.Manifest{
+	Versioned: specs.Versioned{
+		SchemaVersion: 2, // historical value. does not pertain to OCI or docker version
+	},
+	MediaType: ocispec.MediaTypeImageManifest,
+	Config:    content.NewDescriptorFromBytes(ocispec.MediaTypeImageConfig, configLayer),
+	Layers:    []ocispec.Descriptor{manifestLayerDescriptor},
+}
+manifestContent, err := json.Marshal(manifest)
+if err != nil {
+	panic(err)
+}
+manifestDescriptor := content.NewDescriptorFromBytes(ocispec.MediaTypeImageManifest, manifestContent)
+err = repo.PushReference(ctx, manifestDescriptor, bytes.NewReader(manifestContent), tag)
+if err != nil {
+	panic(err)
+}
+fmt.Println("Pushed manifest")
+```
+
+## Fetch the manifest from the repository by tag
+
+The following code snippet demonstrates how to fetch a manifest from the repository by its tag. First, the descriptor associated with the tag is resolved using [(*Repository) Resolve](https://pkg.go.dev/oras.land/oras-go/v2@v2.5.0/registry/remote#Repository.Resolve). Then, the content is fetched using [(*Repository) Fetch](https://pkg.go.dev/oras.land/oras-go/v2@v2.5.0/registry/remote#Repository.Fetch). Finally, the content is read using [ReadAll](https://pkg.go.dev/oras.land/oras-go/v2@v2.5.0/content#ReadAll) from the [content](https://pkg.go.dev/oras.land/oras-go/v2@v2.5.0/content) package.
+
+```
+desc, err := repo.Resolve(ctx, tag)
+if err != nil {
+	panic(err)
+}
+rc, err := repo.Fetch(ctx, desc)
+if err != nil {
+	panic(err)
+}
+defer rc.Close() // don't forget to close
+fetchedManifestContent, err := content.ReadAll(rc, desc)
+if err != nil {
+	panic(err)
+}
+fmt.Println(string(fetchedManifestContent))
```

Comment:
Might be helpful to demonstrate how to parse the fetched manifest content and get the layers from it

Reply:
added a section about this.